### PR TITLE
Add JEI integration for reactor moderators and turbine coils

### PR DESCRIPTION
**Pull request by jeremiahwinsley**

### CODE REVIEW

File: BiggerReactors/build.gradle [hunk 1]
```diff
@@ -145,6 +149,11 @@ dependencies {
 
     compileOnly "mekanism:Mekanism:1.16.3-10.0.11.436:api"
 
+    // compile against the JEI API but do not include it at runtime
+    compileOnly fg.deobf("mezz.jei:jei-${mc_version}:${jei_version}:api")
+    // at runtime, use the full JEI jar
+    runtimeOnly fg.deobf("mezz.jei:jei-${mc_version}:${jei_version}")
```
Comment (RogueLogix) [inline BiggerReactors/build.gradle]: JEI is added in the workspace for runtime

File: BiggerReactors/src/main/java/net/roguelogix/biggerreactors/client/jei/BiggerReactorsJEIPlugin.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        List<FluidModeratorCategory.Recipe> fluidModeratorRecipes = new ArrayList<>();
+        List<BlockModeratorCategory.Recipe> blockModeratorRecipes = new ArrayList<>();
+
+        ReactorModeratorRegistry.getImmutableRegistry().forEach((block, moderatorProperties) -> {
+            if (block instanceof FlowingFluidBlock) {
+                FlowingFluidBlock fluidBlock = (FlowingFluidBlock) block;
+                FluidStack stack = new FluidStack(fluidBlock.getFluid(), 1000);
+                fluidModeratorRecipes.add(new FluidModeratorCategory.Recipe(stack, moderatorProperties));
+            } else if (!(block instanceof AirBlock)) {
```
Comment (RogueLogix) [inline BiggerReactors/src/main/java/net/roguelogix/biggerreactors/client/jei/BiggerReactorsJEIPlugin.java:68]: Air is a valid moderator, and has its own properties, its shit, but it does have them

**Review by RogueLogix — CHANGES_REQUESTED** [review]

File: BiggerReactors/src/main/java/net/roguelogix/biggerreactors/client/jei/BiggerReactorsJEIPlugin.java (hunk 2 above)
Comment (jeremiahwinsley) [reply BiggerReactors/src/main/java/net/roguelogix/biggerreactors/client/jei/BiggerReactorsJEIPlugin.java:68]: Problem is JEI considers an ItemStack containing `minecraft:air` an empty/invalid recipe, and drops it. Not really sure how to work around that other than excluding them entirely.

Comment (RogueLogix) [reply BiggerReactors/src/main/java/net/roguelogix/biggerreactors/client/jei/BiggerReactorsJEIPlugin.java:68]: alll right then, guess there isn't much that can be done about that

**Review by RogueLogix — APPROVED** [review]

**Comment by RogueLogix** [comment]
#39

**Pull request merged**